packages/https-proxy/test/integration/proxy_spec.js

```diff
@@ -90,29 +90,31 @@ describe('Proxy', () => {
     })
   })
 
-  it('closes outgoing connections when client disconnects', function () {
+  it('closes outgoing connections when client disconnects', async function () {
     this.sandbox.spy(net, 'connect')
 
-    return request({
+    await request({
       strictSSL: false,
       url: 'https://localhost:8444/replace',
       proxy: 'http://localhost:3333',
       resolveWithFullResponse: true,
     })
-    .then(() => {
-      // ensure the outgoing socket created for this connection was destroyed
-      expect(net.connect).calledOnce
 
-      const socket = net.connect.getCalls()[0].returnValue
+    // ensure the outgoing socket created for this connection was destroyed
+    expect(net.connect).calledOnce
 
-      return new Promise((resolve) => {
+    const socket = net.connect.getCalls()[0].returnValue
+
+    // sometimes the close event happens before we can attach the listener, causing this test to flake
+    if (!socket.destroyed || !socket.readyState === 'closed') {
+      await new Promise((resolve) => {
```

Comment:
guessing if this promise never resolves the test just times out, right?

Reply:
Yep, times out after two seconds.